lib/ecto/repo/queryable.ex

```diff
@@ -36,6 +36,20 @@ defmodule Ecto.Repo.Queryable do
     one!(repo, adapter, query_for_get(queryable, id), opts)
   end
 
+  def get_by(repo, adapter, queryable, clause, opts) do
+    [key] = Keyword.keys(clause)
+    [value] = Keyword.values(clause)
+
+    one(repo, adapter, query_for_get_by(queryable, key, value), opts)
+  end
+
+  def get_by!(repo, adapter, queryable, clause, opts) do
+    [key] = Keyword.keys(clause)
```

Comment:
I tried some crazy things to pattern match this clause, but not the easiest one. :smile: 

I have to think a little before answering this. I will try as soon as I can. :)